### DISCUSSION TITLE
Wrapping long words in Captions

### DISCUSSION THREAD

**Discussion by fjahn** [ImageMagick/ImageMagick#4227] [Help] [answered]
I want to generate images with long words inside them, like this:
`convert -pointsize 90 -size 400x250 caption:"Looooooooooooong Word" img.png `
Result:
![img](https://user-images.githubusercontent.com/74260729/133406354-356b6eef-abbd-46de-b4e3-a4a5e5130d23.png)

Usually, ImageMagick can wrap _between_ words, but does not wrap _within_ words, even if this means that the word will be clipped. Curiously, in this case, even the second word is not shown on another line, even though it should be wrapped.

**I would like the word to be wrapped mid-word if there is no room to avoid clipping otherwise.** Is this possible somehow? I have already looked through the [documentation for Text Handling](https://legacy.imagemagick.org/Usage/text/) and found nothing, but I wanted to make sure it's not possible before giving up.

**Comment by snibgo** [comment]
What version of IM, on what platform?

With v6.9.9-50 on Windows 8.1, I get:
![i](https://user-images.githubusercontent.com/33812368/133440083-f7ea873c-a564-4ecc-b9c5-5feaf42d110c.png)

IM won't break words.

**Reply by fjahn** [reply]
> What version of IM, on what platform?

I am running ImageMagick 7.1.0-6 on MacOS 11.4 (Apple Silicon).

> IM won't break words.

Okay, that's too bad but thanks for the response!

**Comment by fmw42** [comment]
On IM 6.9.12.22 or 7.1.0.7 Q16 Mac OSX, I get clipping and no wrapping.

`convert -pointsize 90 -size 400x250 caption:"Looooooooooooong Word" img.png `

![img](https://user-images.githubusercontent.com/13458218/133458208-e6d7e7a7-312f-489a-931c-4ebfc4761d5f.png)

**Comment by urban-warrior** [comment, marked as answer]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick sometime this week.

To split words that exceed the canvas size, try `-define caption:split=true`.

**Reply by fjahn** [reply]
`convert -pointsize 90 -size 400x300 -define caption:split=true caption:"Looooooooong Word" img.png` produces the following, which is exactly what I wanted, thank you!

![img](https://user-images.githubusercontent.com/74260729/133961693-c8604222-efa8-45a4-a914-b8d19e65ce1e.png)
